src/ui/public/courier/saved_object/saved_object.js

```diff
@@ -18,6 +18,28 @@ import MappingSetupProvider from 'ui/utils/mapping_setup';
 
 import DocSourceProvider from '../data_source/admin_doc_source';
 import SearchSourceProvider from '../data_source/search_source';
+import { getTitleAlreadyExists } from './get_title_already_exists';
+
+/**
+ * An error message to be used when the user rejects a confirm overwrite.
+ * @type {string}
+ */
+const OVERWRITE_REJECTED = 'Overwrite confirmation was rejected';
+/**
+ * An error message to be used when the user rejects a confirm save with duplicate title.
+ * @type {string}
+ */
+const SAVE_DUPLICATE_REJECTED = 'Save with duplicate title confirmation was rejected';
+
+/**
+ * Returns true if the error is expected and should be suppressed (e.g. not shown to the user).
+ * @param error {Error} the error
+ * @return {boolean}
+ */
+function getShouldSuppressExpectedError(error) {
```

Comment:
I think that's a little confusing too because it's not an error at all, it's a user response.  The whole flow is a bit awkward, and it'd be better if we just stopped using the confirmModalPromise to begin with, but I don't want to do that refactor now.

 I have no better suggestion, so sgtm for now.